star-ros2/src/star_perception/star_perception/hailo_runner.py

```
@@ -0,0 +1,390 @@
+
+def letterbox_image(rgb: np.ndarray,
+                    params: LetterboxParams) -> np.ndarray:
+    """
+    Resize and pad rgb into params.target x params.target uint8 RGB.
+
+    Uses nearest-neighbour resize via numpy (avoids cv2 dependency in
+    the unit tests). Production callers may swap this for cv2.resize
+    with INTER_LINEAR for better quality, at the cost of a runtime dep.
+    """
+    if rgb.ndim != 3 or rgb.shape[2] != 3:
+        raise ValueError("Expected HxWx3 RGB image.")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)